### DISCUSSION TITLE
Read() operation

### DISCUSSION THREAD

**Discussion by RahulMvr2824** [kenjdavidson/react-native-bluetooth-classic#271] [Q&A]
I'm performing a device.read() operation to fetch data from the bluetooth module. As I understood this operation works like First in First Out which means the message that comes first will be read first. But I had a use case where I want Last in First out method. Is there any way for that process using device.read().

Thanks in Advance!!

**Comment by kenjdavidson** [comment]
With the standard functionality no, there is not.  With that said, you have a number of options:

1. You can create your own `ConnectionImpl` which extends from the standard `DelimitedStringConnectionImpl` and overwrites the read method to return the last message instead of the first message.  You then need to overwride the standard native module creation.  I believe there is a documentation page on how to do this.
2. If you think this is standard enough functionality, you can introduce a connection parameter for the `DelimitedStringConnectionImpl` that when set would return the buffer in a LIFO (instead of FIFO) order, I'd accept a PR for this.

These are the only options at this point in time.
